packages/backend-data/src/convert_schema.ts

```diff
@@ -16,22 +29,63 @@ export const isModelSchema = (
   return (
     schema !== null &&
     typeof schema === 'object' &&
+    'transform' in schema &&
     typeof schema.transform === 'function'
   );
 };
 
+/**
+ * Determine if the input schema is a collection of model schemas, and perform type narrowing.
+ * @param schema the schema that might be a collection of model schemas
+ * @returns a boolean indicating whether the schema is a collection of derived model schema, with type narrowing
+ */
+export const isCombinedSchema = (
+  schema: DataSchemaInput
+): schema is DerivedCombinedSchema => {
+  return (
+    schema !== null &&
+    typeof schema === 'object' &&
+    'schemas' in schema &&
+    Array.isArray(schema.schemas)
+  );
+};
+
+/**
+ * SQL provision strategy requires a unique name per backend that is consistent between deployments
+ * @param scope construct scope
+ * @returns string provision strategy name
+ */
+const sqlProvisionStrategyName = (scope: Construct): string =>
+  `${scope.node.getContext(
+    CDKContextKey.DEPLOYMENT_TYPE
+  )}${scope.node.getContext(
+    CDKContextKey.BACKEND_NAMESPACE
+  )}${scope.node.getContext(CDKContextKey.BACKEND_NAME)}`;
```

Comment:
are there any characters that are not allowed in the strategy name? The backend_name is the branch name for branch deployments which has a wide range of valid characters. Similarly for the backend_namespace; in sandbox deployments that is the package.json#name value which may have `@/_-` or other symbols.

Another thought: does this string need to be human readable? If not, maybe we should hash this value so that 1) it's normalized to a known set of characters and 2) it discourages anyone trying to parse it

Reply:
Great catch. We use this value (among other things) for naming the Lambda resolvers that are generated for the SQL schema, so many special characters like slashes and `@` symbols, etc. would cause the deployment to fail. 

The value doesn't need to be human-readable, so I agree that a hash is going to be safest here. I'll make that change.

Reply:
Seems like we can either re-use [`BackendIndentifierConversions.toStackName()`](https://github.com/aws-amplify/amplify-backend/blob/main/packages/platform-core/src/backend_identifier_conversions.ts#L54-L75) as is, or add another `static` method to the class for this purpose. Would that make sense here, @edwardfoyle?

Reply:
As long as dashes are okay characters, `.toStackName` could work. However, I'm not sure I want to couple that to using it here in a place that's not a stack name. I think my most preferred option is in [this comment](https://github.com/aws-amplify/amplify-backend/pull/1213#discussion_r1556491179). We could inject something like a `StableBackendHashGetter` so that this code doesn't have to deal with parsing CDK context at all

Reply:
Thanks. Will add the preferred option shortly

Reply:
Updated